projects/Rockchip/devices/OdroidGoAdvance/packages/odroidgoa-utils/sources/oga_events.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+
+import evdev
+import asyncio
+import os.path
+from subprocess import check_output
+
+pwrkey = evdev.InputDevice("/dev/input/event0")
+
+if os.path.exists("/dev/input/by-path/platform-odroidgo3-joypad-event-joystick"):
+    odroidgo_joypad = evdev.InputDevice("/dev/input/by-path/platform-odroidgo3-joypad-event-joystick")
+    devicename = "GO-Super Gamepad"
+    odroidgo_volume = evdev.InputDevice("/dev/input/by-path/platform-odroidgo3-keys-event")
+else:
+    odroidgo_joypad = evdev.InputDevice("/dev/input/by-path/platform-odroidgo2-joypad-event-joystick")
+    devicename = "GO-Advance Gamepad"
+
+need_to_swallow_pwr_key = False # After a resume, we swallow the pwr input that triggered the resume
+
+class Power:
+    pwr = 116
+
+class Joypad:
+    l1 = 310
+    r1 = 311
+
+    up = 544
+    down = 545
+    left = 546
+    right = 547
+
+    f1 = 704
+    f2 = 705
+    f5 = 708
+    volu = 115
+    vold = 114
+
+def runcmd(cmd, *args, **kw):
+    print(f">>> {cmd}")
+    check_output(cmd, *args, **kw)
+
+async def handle_event(device):
+    async for event in device.async_read_loop():
+        global need_to_swallow_pwr_key
+        if device.name == "rk8xx_pwrkey":
+            keys = odroidgo_joypad.active_keys()
```

Comment:
Unless I'm mistaken you don't really need this for the power key to function as a power key. The reason it doesnt do anything is cause Lakka has power sleep disabled globally. But if you inspec the /dev/input devices you will see that it is properly mapped.